engine/CMakeLists.txt

```diff
@@ -808,9 +808,9 @@ IF (USE_SYSTEM_BOOST)
         MESSAGE("-- Boost_1_67_Or_Later_Result: ${Boost_1_67_Or_Later_Result}")
         MESSAGE("-- USE_PYTHON_3: ${USE_PYTHON_3}")
         IF (Boost_1_67_Or_Later_Result)
-            SET(BOOST_PYTHON_COMPONENT "python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
+            SET(BOOST_PYTHON_COMPONENT "python38")
```

Comment:
Why tie Boost-Python to a particular Python version, on a particular set of platforms/variants? Please revert this, or find another way to do it. These `python38` changes are probably why the build is breaking.